docs/source/quickstart.mdx

```diff
@@ -0,0 +1,405 @@
+
+```
+npm install apollo-server @apollo/gateway
+```
+
+Then paste the following into an `index.js` file as a minimal (not-yet-functional) gateway implementation:
+
+```js{4,7}:title=index.js
+const { ApolloServer } = require('apollo-server');
+const { ApolloGateway } = require('@apollo/gateway');
+
+const supergraphSchema = ''; // TODO!
+
+const gateway = new ApolloGateway({
+  supergraphSdl: supergraphSchema
+});
+
+const server = new ApolloServer({
+  gateway,
+  // Subscriptions are not currently supported in Apollo Federation
+  subscriptions: false
+});
+
+server.listen().then(({ url }) => {
+  console.log(`🚀 Gateway ready at ${url}`);
+}).catch(err => {console.error(err)});
+```
+
+This code demonstrates the basic flow for creating a gateway:
+
+1. We initialize an `ApolloGateway` instance and pass it the complete composed schema for our supergraph.
+    * _Note the `TODO`. We'll obtain the composed schema in the next section._
+
+2. We initialize an `ApolloServer` instance and pass it our gateway via the `gateway` option.
+    * _We also disable `subscriptions` because they aren't yet supported in Apollo Federation._
+
+3. We call `listen` on our server instance to begin listening for incoming requests.
+
+If we run this code as-is with `node index.js`, we get an error:
+
+```
+This data graph is missing a valid configuration. Syntax Error: Unexpected <EOF>.
+```
+
+That's because our `supergraphSchema` is currently empty! Next, we'll compose that schema.
+
+## 3. Compose the supergraph schema
+
+As a best practice, your gateway should _not_ be responsible for composing its own supergraph schema. Instead, a separate process should compose the schema and provide it to the gateway. This helps improve reliability and reduce downtime when you make changes to a subgraph.
+
+There are multiple ways to compose a supergraph schema from our subgraph schemas:
+
+* On our local machine using the Rover CLI (we'll start with this)
+* Via managed federation in Apollo Studio (we'll switch to this in a later section)
+
+Using managed federation is strongly recommended for production environments. We'll start with local composition just to get up and running.
+
+### Obtaining subgraph details
+
+To compose our supergraph schema, we need the following information about each of our subgraphs:
+
+* The subgraph's schema
+* The URL of the subgraph's GraphQL endpoint (which must be accessible by the gateway)
+
+We'll start with just two subgraphs. Because we're using Apollo-hosted example services for our subgraphs, we know their endpoint URLs. We _don't_ have their schemas, however.
+
+#### Fetching subgraph schemas
+
+We can fetch a running subgraph's schema by executing an enhanced introspection query on it. The Rover CLI provides a command to do exactly that.
+
+From your project directory, run the following command in your terminal:
+
+```shell
+rover subgraph introspect https://7bssbnldib.execute-api.us-east-1.amazonaws.com/Prod/graphql
```

Comment:
This is where people will get stuck if they're running node 15. Installing rover still works but you get stuck with v0.0.3 which doesn't support introspect, so you get an error here.

Reply:
Presumably everyone who doesn't already have node <15 will get stuck here, because installing Node means installing 15 today.

Reply:
Something I didn't know until very recently is that one should only every use even-numbered versions of Node.js to maximize compatibility, so I think we'll be fine with mentioning at the top of the tutorial that it requires Node 12 or 14 (helpfully, the only two versions of Node that are in "active LTS")